src/ImageSharp/Formats/Png/PngEncoder.cs

```diff
@@ -18,7 +17,7 @@ public class PngEncoder : QuantizingImageEncoder
 
         // We set the quantizer to null here to allow the underlying encoder to create a
         // quantizer with options appropriate to the encoding bit depth.
-        this.Quantizer = null;
+        this.Quantizer = null!;
```

Comment:
If `QuantizingImageEncoder.Quantizer == null` is a valid state, shouldn't it be nullable?

Reply:
hmm yes, but `QuantizingImageEncoder.Quantizer` is also used in encoders in many other formats. In order not to change more code, I suppressed the nullable notification here.

Reply:
It shouldn't be nullable anywhere really, however we're unable to lazily set the quantizer to the required bitdepth without it being nullable here. Best just leaving nullability checks disabled in this file for now. The core encoder deals with nullability. 

I want to have a good look at the quantizers again for V4.

Reply:
Then we can disable nullable in the `QuantizingImageEncoder.cs`. This allows us to use nullable in the `Encoder` and avoid property `Quantizer`'s null checking.

Reply:
You can't do that because that breaks the ability to use a palette quantizer in the presence of a color palette in the metadata #2455  If someone explicitly sets a quantizer they are showing intent.

Reply:
I am sorry. I will revert the enable disable in `PngEncoder.cs`.

Reply:
Nothing to apologize for! 🙂